src/bio.c

```diff
@@ -308,36 +295,20 @@ void *bioProcessBackgroundJobs(void *arg) {
             serverPanic("Wrong job type in bioProcessBackgroundJobs().");
         }
         zfree(job);
-
-        /* Lock again before reiterating the loop, if there are no longer
-         * jobs to process we'll block again in pthread_cond_wait(). */
-        pthread_mutex_lock(&bwd->bio_mutex);
-        listDelNode(bwd->bio_jobs, ln);
-        bio_jobs_counter[job_type]--;
-        pthread_cond_signal(&bwd->bio_newjob_cond);
+        atomic_fetch_sub(&bio_jobs_counter[job_type], 1);
     }
 }
 
 /* Return the number of pending jobs of the specified type. */
 unsigned long bioPendingJobsOfType(int type) {
-    bio_worker_data *const bwd = &bio_workers[bio_job_to_worker[type]];
-
-    pthread_mutex_lock(&bwd->bio_mutex);
-    unsigned long val = bio_jobs_counter[type];
-    pthread_mutex_unlock(&bwd->bio_mutex);
-
-    return val;
+    return atomic_load(&bio_jobs_counter[type]);
 }
 
 /* Wait for the job queue of the worker for jobs of specified type to become empty. */
-void bioDrainWorker(int job_type) {
-    bio_worker_data *const bwd = &bio_workers[bio_job_to_worker[job_type]];
-
-    pthread_mutex_lock(&bwd->bio_mutex);
-    while (listLength(bwd->bio_jobs) > 0) {
-        pthread_cond_wait(&bwd->bio_newjob_cond, &bwd->bio_mutex);
+void bioDrainWorker(int type) {
+    while (bioPendingJobsOfType(type) > 0) {
+        usleep(100); /* Sleep for 1ms and check again*/
```

Comment:
```suggestion
        usleep(100);
```
The comment is out of date, but also I don't think the comment is telling us anything.

Reply:
Removed the outdated comment